packages/docusaurus/src/server/plugins/index.ts

```diff
@@ -5,21 +5,51 @@
  * LICENSE file in the root directory of this source tree.
  */
 
+import {DocusaurusConfig} from '../config';
+import {LoadContext} from '..';
+import {RouteConfig} from '../routes';
+
 import fs from 'fs-extra';
 import importFresh from 'import-fresh';
 import path from 'path';
 import {generate} from '@docusaurus/utils';
-import {LoadContext} from '..';
+
+export interface Plugin<T> {
+  name: string;
+  loadContent?(): T;
+  contentLoaded?({
+    content: T,
+    actions: DocusaurusPluginContentLoadedActions,
+  }): void;
+  postBuild?(props: any): void;
```

Comment:
Is this type `Props`? Not very familiar with this part.

Reply:
i think so.

Reply:
Ok added more types. I'm glad we have less `any` now.